### ISSUE TITLE
json rpc parse error with converted conf file

### ISSUE THREAD

**Issue by jonathanrichardson** [spdk/spdk#1741]
I converted my old style conf file using config_converter.py (described in steps to reproduce below). There is no Nvme section in the json file output. When I run spdk I get an error saying json rpc parsing failed. There is an iscsi subsection in the conf file with a iscsi_set_options method that causes the error.

I had an Rpc and Transport section in my conf file but the script said to remove it so I did.

## Expected Behavior

The output from the config_converter.py script should not generate a json file that can't be parsed. 

## Current Behavior

The following errors occur when running spdk with -c option.

[2021-01-11 20:41:39.448690] json_config.c: 392:app_json_config_load_subsystem_config_entry: *DEBUG*: 	params: {
            "allow_duplicated_isid": false,
            "default_time2retain": 20,
            "mutual_chap": false,
            "max_connections_per_session": 2,
            "nop_timeout": 60,
            "disable_chap": false,
            "require_chap": false,
            "node_base": "iqn.2016-06.io.spdk",
            "chap_group": 0,
            "max_sessions": 128,
            "immediate_data": true,
            "error_recovery_level": 0,
            "nop_in_interval": 30,
            "default_time2wait": 2,
            "max_queue_depth": 64,
            "first_burst_length": 8192
          }
[2021-01-11 20:41:39.793791] jsonrpc_client.c:  95:jsonrpc_parse_response: *DEBUG*: JSON string is :
{"jsonrpc":"2.0","id":0,"error":{"code":-32601,"message":"Method not found"}}

[2021-01-11 20:41:39.793809] json_config.c: 219:rpc_client_poller: *ERROR*: error response: 
{
  "code": -32601,
  "message": "Method not found"
}
[2021-01-11 20:41:39.794297] json_config.c: 135:app_json_config_load_done: *DEBUG*: Config load finished with rc -22

## Steps to Reproduce

1. Create temp.conf file with the following contents:
[Global]
  LogFacility "local7"

[Nvmf]
  AcceptorPollRate 10000

[Nvme]
  TransportId "trtype:PCIe traddr:0007:01:00.0" Nvme0
  HotplugEnable Yes
  HotplugPollRate 1000000

[Subsystem0]
  NQN nqn.2016-06.io.spdk:cnode0
  AllowAnyHost Yes
  Listen RDMA 192.168.1.10:1023
  SN SPDK00000000000001
  Namespace Nvme0n1

2. Convert the script to the new json format using the command:
cat temp.conf | ./config_converter.py > temp.json
3. Run spdk: nvmf_tgt -m 0xf -c temp.json

## Context (Environment including OS version, SPDK version, etc.)
Running on arm64 platform using spdk 21.01 pre release at commit:

commit 731cd47e66dceeedd07381e469f55ebbe9773e91 (HEAD -> master, origin/master, origin/HEAD)
Author: Michal Berger <michalx.berger@intel.com>
Date:   Tue Dec 29 11:54:05 2020 +0100

    test/common: Add SPDK_APP wrapper for spdk_tgt

**Comment by tomzawadzki** [comment]
Hi @jonathanrichardson Please see patch https://review.spdk.io/gerrit/c/spdk/spdk/+/5879 and let me know if it fixes this for you.

Just a bit of background. The `config_converter.py` cannot account for different SPDK applications and configuration options when building SPDK. You ran into this issue due to `iscsi_set_options` not being supported in `nvmf_tgt`, but working in other apps like `spdk_tgt`. The patch above changes `config_converter.py` to only output configuration present in legacy config file.

**Comment by jonathanrichardson** [comment]
OK thanks. I figured the section was invalid so I removed it but wanted to make sure I wasn't missing something. That does solve the problem with the iscsi section.

When I run spdk now with the json config it says there is no transport, but it had told me to remove the Transport section in my old style conf file. Can you tell me what else is missing, and if this is considered a bug or there is something I need to manually add to my json file:

[2021-01-12 20:39:18.858204] accel_engine.c: 692:spdk_accel_engine_initialize: *NOTICE*: Accel engine initialized to use software engine.
[2021-01-12 20:39:18.939414] nvmf.c: 653:spdk_nvmf_tgt_listen_ext: *ERROR*: Unable to find RDMA transport. The transport must be created first also make sure it is properly registered.
[2021-01-12 20:39:18.943412] json_config.c: 219:rpc_client_poller: *ERROR*: error response: 
{
  "code": -32602,
  "message": "Invalid parameters"
}

**Comment by jonathanrichardson** [comment]
I found the JSON RPC doc and noticed the sections that were missing in my json file. I wasn't aware it corresponded to the existing rpc interface. I added the nvmf_create_transport method to the nvmf subsystem, and bdev_nvme_attach_controller to the bdev subsystem and it starts fine. There are probably use cases with the other apps I haven't looked at before so not sure it is reasaonble to request that the converter script add these to the json file or not. Not adding sections though is better than adding and invalid one like iscsi. Thanks!
